test/spec/Feature/Query/RpcSpec.hs

```diff
@@ -970,6 +970,29 @@ spec actualPgVersion =
             , matchHeaders = ["Content-Type" <:> "text/plain; charset=utf-8"]
             }
 
+        it "can get raw xml output with Accept: text/xml" $
+          request methodGet "/rpc/return_scalar_xml" (acceptHdrs "text/xml") ""
+            `shouldRespondWith`
+            "<my-xml-tag/>"
+            { matchStatus = 200
+            , matchHeaders = ["Content-Type" <:> "text/xml; charset=utf-8"]
+            }
+
+        it "should fail with Accept: text/xml" $
+          request methodGet "/rpc/welcome" (acceptHdrs "text/xml") ""
+            `shouldRespondWith`
+            [json|
+              {
+                "hint":"No function matches the given name and argument types. You might need to add explicit type casts.",
+                "details":null,
+                "code":"42883",
+                "message":"function xmlagg(text) does not exist"
+              }
+            |]
+            { matchStatus = 404
+            , matchHeaders = ["Content-Type" <:> "application/json; charset=utf-8"]
+            }
+
```

Comment:
Hm. I'm not sure how, but I think we should answer with `406 Not Acceptable` and not `404 Not Found` in this case. It's not that we can't find `/rpc/welcome` - but we can't satisfy the `Accept: text/xml`.

Reply:
@wolfgangwalther: That goes beyond my "pattern-recognition" haskell abilities. Can you help?

Reply:
Hm. Maybe we don't need to throw that error at all. Do we actually need `xmlagg()`? Or could we just use `string_agg()` with an explicit cast to `text`? xmlagg is not doing anything other that concatenating, right? And we'll just return text anyway.

That would allow returning non-xml with xml header, though. So maybe we just do `string_agg(col::pg_catalog.xml::text)`. On a real xml column, the first cast would be a no-op. But with a different column type (or invalid xml content for that matter) it would throw an error.

Reply:
>  xmlagg is not doing anything other that concatenating, right?

Correct.

> That would allow returning non-xml with xml header, though. 

Yes, unfortunately. For example: I think a text of "four < eight" is not valid XML, the "<" would have to be escaped.

Reply:
> Yes, unfortunately. For example: I think a text of "four < eight" is not valid XML, the "<" would have to be escaped.

Yes, but I think my suggestion of `string_agg(col::pg_catalog.xml::text)` would prevent that.

Reply:
Thanks, now I understand. I have committed the changes.

Reply:
How about just catching the `function xmlagg(*) does not exist` error in:

https://github.com/PostgREST/postgrest/blob/057e8d63bfa351e5ba0518109dd9058aa113da4b/src/PostgREST/Error.hs#L250-L254

The 2nd parameter(`m`) is the message I believe - then you can map that to a 406 status code.

I also think we should go back to `xmlagg` instead of `stringagg`, the strictness will likely lead to better error messages for malformed xml.

Reply:
The above would be the simpler solution I think. Note that `octet-stream` and `text/plain` are also giving a 400 error instead of a 406 currently. 

The ideal way to solve all of these issues is to add more smartness to our internal cache(DbStructure). Likely by detecting that the function `returns xml` before executing it <- but this will be more complicated and can be done in a later PR.